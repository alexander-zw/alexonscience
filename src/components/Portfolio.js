/**
 * This component contains my portfolio. It's kind of more of a resume but
 * portfolio sounds fancy.
 * 
 * This page is for anyone who came to learn about me personally, including
 * who I am, my skills and interests, and some of my projects that can't be
 * accessed directly on my website.
 */
import React from 'react';
import '../styles/index.css'
import '../styles/Portfolio.css';

function Portfolio() {
    return (
        <div className="text-div top-margin bottom-margin">
            <span className="name-title">Alexander Wu</span>

            <p>Email: alexwu68 [at] berkeley [dot] edu</p>
        </div>
    );
}

export default Portfolio;

/*
import React from 'react';
import '../styles/index.css'
import '../styles/Portfolio.css';

function Portfolio() {
    return (
        <div>
            <nav class="section-nav">
                <ol>
                    <li><a href="#introduction">Introduction</a></li>
                    <li><a href="#request-response">Request &amp; Response</a></li>
                    <li><a href="#authentication">Authentication</a></li>
                …
                    <li class=""><a href="#filters">Filters</a></li>
                </ol>
            </nav>

            <div className="text-div top-margin bottom-margin">
                <span className="name-title">Alexander Wu</span>

                <p>Email: alexwu68 [at] berkeley [dot] edu</p>

                <h1>Smooth Scrolling Sticky ScrollSpy Navigation</h1>
                <section id="introduction">
                    <h2>Introduction</h2>
                    <p>…</p>
                </section>
                <section id="request-response">
                    <h2>Request &amp; Response</h2>
                    <p>…</p>
                </section>
                <section id="authentication">
                    <h2>Authentication</h2>
                    <p>…</p>
                </section>
                …
                <section id="filters">
                    <h2>Filters</h2>
                    <p>…</p>
                </section>
            </div>
        </div>
    );
}

export default Portfolio;

*/
