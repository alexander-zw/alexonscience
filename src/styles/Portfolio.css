.outer-container {
    display: grid;
    grid-template-columns: 20% auto;
    grid-template-areas: "nav content";

}

.contents-nav {
    position: -webkit-sticky; /* Safari earlier versions. */
	position: sticky;
	top: 3.2em;
	align-self: start;
}

.contents-ol {
    list-style-type: none;
}

.contents-li {
    margin-top: 1em;
}

.contents-item {
    text-decoration: none;
    color: #636363;
}

.contents-item:hover {
    color: #008388;
}

.name-title {
    padding-top: 0.2em;
    font-size: 3em;
    font-family: 'Oleo Script', 'Georgia', serif;
    background-image: linear-gradient(to right, #249141, #249c8a, #079cb3, #3464eb);
    -webkit-background-clip: text;
    -moz-background-clip: text;
    background-clip: text;
    color: transparent;
}

.section {
    padding-bottom: 10em;
    position: relative;
    /* This allows linking via table of contents to arrive at the right place. */
    padding-top: 2em;
    top: -3em;
}
