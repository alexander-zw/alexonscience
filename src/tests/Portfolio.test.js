/**
 * Test for the Portfolio.js component.
 */
import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Portfolio from '../components/Portfolio';

it('renders correctly', () => {
    const tree = renderer.create(<Portfolio />).toJSON();
    expect(tree).toMatchSnapshot();
});
