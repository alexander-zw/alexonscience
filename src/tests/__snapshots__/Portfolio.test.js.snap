// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="text-div top-margin bottom-margin"
>
  <span
    className="name-title"
  >
    Alexander Wu
  </span>
  <p>
    Email: alexwu68 [at] berkeley [dot] edu
  </p>
</div>
`;
